ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,377 @@
+import numbers
+from bisect import bisect_right
+from typing import Any, Callable, List, Tuple, Union
+
+from ignite.engine import CallableEventWithFilter, Engine, Events, EventsList
+from ignite.handlers import BaseParamScheduler
+
+
+class StateParamScheduler(BaseParamScheduler):
+    """An abstract class for updating an engine state parameter values during training.
+
+    Args:
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Note:
+        Parameter scheduler works independently of the internal state of the attached engine.
+        More precisely, whatever the state of the engine (newly created or used by another scheduler) the scheduler
+        sets defined absolute values.
+
+    .. versionadded:: 0.6.0
+
+    """
+
+    def __init__(
+        self, param_name: str, save_history: bool = False,
+    ):
+        super(StateParamScheduler, self).__init__(param_name, save_history)
+
+    def attach(
+        self,
+        engine: Engine,
+        event: Union[str, Events, CallableEventWithFilter, EventsList] = Events.ITERATION_COMPLETED,
+    ) -> None:
+        """Attach the handler to engine. After the handler is attached, the ``Engine.state`` will add a new attribute
+        with name ``param_name``. Then, current parameter value can be retrieved by from ``Engine.state`` when the
+        engine runs.
+
+        Args:
+            engine: trainer to which the handler will be attached.
+            event: trigger ``param_name`` value update.
+
+        """
+        if hasattr(engine.state, self.param_name):
+            raise ValueError(
+                f"Attribute: '{self.param_name}' is already in Engine.state and might be "
```

Comment:
```
Attribute: '{self.param_name}' already defined in the Engine.state. This may be a conflict between multiple StateParameterScheduler handlers. Please choose another name.
```